src/backends/InteractiveBackend.cpp

```diff
@@ -2,6 +2,44 @@
 
 #include "impls.h"
 
+size_t make_left(const char* arr, size_t pos) {
+    size_t res = 1;
+    --pos;
+    while (pos) {
+        if ((arr[pos] & 0xc0) == 0x80) {
+            ++res;
+        } else {
+            break;
+        }
+        --pos;
+    }
+    return res;
+}
+size_t make_right(const char* arr, size_t pos) {
+    size_t res = 1;
+    ++pos;
+    if ((arr[pos] & 0xc0) == 0x80) {
+        while ((arr[pos] & 0xc0) == 0x80) {
+            ++res;
+            ++pos;
+        }
+    }
+    return res;
+}
+
+size_t utf8_length(const char* str, size_t siz = -1) {
```

Comment:
I tried this library and changed my mind, because it checks every character and throws exceptions on non-valid input and that needs to be handled. I have no idea how to handle such things in a separate thread for console reading and whether they make sense in the console. That's why I decided to improve the current functions so that they just return values that will definitely not lead to an out of range. 

If it is not suitable, I can redo it.

Reply:
Any news?